src/services/getFrameKmsService.ts

```diff
@@ -94,211 +108,264 @@ async function isLTEServiceEnabled() {
-                const nextState = {status: LteFkmStatus.COMMIT_SUCCESS};
-                await fkmStorage.setItem(framekm, JSON.stringify(nextState));
-            } else if (status === LteFkmStatus.COMMIT_SUCCESS) {
-                console.log(`Done ${framekm}`);
-                fs.unlinkSync(path.join(FRAMEKM_ROOT_FOLDER, framekm));
-                fs.unlinkSync(path.join(METADATA_ROOT_FOLDER, framekm) + '.json');
-                await fkmStorage.del(framekm);
-            }
-        } catch (e) {
-            console.log('Frame ', framekm, ' failed with error ', e);
-            // FetchErrors indicate something is wrong with the network, so we should stop processing
-            // and try to recover LTE in the main loop
-            if (e instanceof FetchError) {
-                throw e;
-            }
-        }
+  for (const framekm of framekms) {
+    try {
+      const { status } = JSON.parse(await fkmStorage.getItem(framekm));
+      console.log(`Processing ${framekm} with status ${status}`);
+      if (status === LteFkmStatus.INIT) {
+        console.log(`Requesting ${framekm}`);
+        const res = await postFrameKmRequest(framekm);
+        const nextState = {
+          status: LteFkmStatus.REQUEST_SUCCESS,
+          signedContentUrls: res.signedContentUrls,
+          id: res.id,
+        };
+        await fkmStorage.setItem(framekm, JSON.stringify(nextState));
+      } else if (status === LteFkmStatus.REQUEST_SUCCESS) {
+        console.log(`Uploading ${framekm}`);
+        const { signedContentUrls, id } = JSON.parse(
+          await fkmStorage.getItem(framekm),
+        );
+        const multiparts = await uploadFrameKmToS3(
+          signedContentUrls,
+          id,
+          path.join(FRAMEKM_ROOT_FOLDER, framekm),
+          framekm,
+          DEFAULT_PART_SIZE,
+        );
+        const nextState = {
+          status: LteFkmStatus.UPLOAD_SUCCESS,
+          id,
+          multiparts,
+        };
+        await fkmStorage.setItem(framekm, JSON.stringify(nextState));
+      } else if (status === LteFkmStatus.UPLOAD_SUCCESS) {
+        console.log(`Committing ${framekm}`);
+        const { id, multiparts } = JSON.parse(
+          await fkmStorage.getItem(framekm),
+        );
+        await commitFrameKm(id, multiparts);
+        const nextState = { status: LteFkmStatus.COMMIT_SUCCESS };
+        await fkmStorage.setItem(framekm, JSON.stringify(nextState));
+      } else if (status === LteFkmStatus.COMMIT_SUCCESS) {
+        console.log(`Done ${framekm}`);
+        const fkm_path = path.join(FRAMEKM_ROOT_FOLDER, framekm);
+        const stats = fs.statSync(fkm_path);
+        const fileSizeInBytes = stats.size;
+        const dateISOString = new Date().toISOString();
+        const dateKey = TOTAL_DATA_UPLOADED_TODAY+dateISOString.split('T')[0];
+        const currentDataUploaded = parseInt(
+          (await getConfig(dateKey)) || '0',
+        );
+        await setConfig(
```

Comment:
I'm worried this will pollute the config table over time with new keys. Maybe it would be better to create a new table for lte-upload-stats. You can use a new SqlStorage similar to `fkmStorage`

Reply:
this is why I have a function to `clearOlderUploadSizes` function that clears them everytime dashcam turns on

Reply:
this way I will have either one or two values only

Reply:
Nevermind, I see they get cleared on startup